runtime/vm/ValueTypeHelpers.cpp

```diff
@@ -69,4 +70,115 @@ isClassRefQtype(J9Class *cpContextClass, U_16 cpIndex)
 {
 	return VM_ValueTypeHelpers::isClassRefQtype((J9ConstantPool *) cpContextClass->ramConstantPool, cpIndex);
 }
+UDATA
+findIndexInFlattenedClassCache(J9FlattenedClassCache *flattenedClassCache, J9ROMNameAndSignature *nameAndSignature)
+{
+        /* first field indicates the number of classes in the cache */
+        UDATA index = UDATA_MAX;
+        UDATA length = flattenedClassCache->numberOfEntries;
+        J9ROMFieldShape *fccEntryField = NULL;
+
+        for (UDATA i = 0; i < length; i++) {
+                fccEntryField = J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, i)->field;
+                if (J9UTF8_EQUALS(J9ROMNAMEANDSIGNATURE_NAME(nameAndSignature), J9ROMFIELDSHAPE_NAME(fccEntryField))
+                        && J9UTF8_EQUALS(J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSignature), J9ROMFIELDSHAPE_SIGNATURE(fccEntryField))
+                ) {
+                        index = i;
+                        break;
+                }
+        }
+        return index;
+}
+
+J9Class *
+findJ9ClassInFlattenedClassCache(J9FlattenedClassCache *flattenedClassCache, U_8 *className, UDATA classNameLength)
+{
+        /* first field indicates the number of classes in the cache */
+        UDATA length = flattenedClassCache->numberOfEntries;
+        J9Class *clazz = NULL;
+
+        for (UDATA i = 0; i < length; i++) {
+                J9UTF8* currentClassName = J9ROMCLASS_CLASSNAME(J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, i)->clazz->romClass);
+                if (J9UTF8_DATA_EQUALS(J9UTF8_DATA(currentClassName), J9UTF8_LENGTH(currentClassName), className, classNameLength)) {
+                        clazz = J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, i)->clazz;
+                        break;
+                }
+        }
+
+        Assert_VM_notNull(clazz);
+        return clazz;
+}
+
+UDATA
+getFlattenableFieldOffset(J9Class *fieldOwner, J9ROMFieldShape *field)
+{
+        Assert_VM_notNull(fieldOwner);
+        Assert_VM_notNull(field);
+
+        UDATA fieldOffset = 0;
+        J9FlattenedClassCache *flattenedClassCache = NULL;
+        J9ROMNameAndSignature *nameAndSig;
+        UDATA fieldIndex = 0;
+
+        nameAndSig = &field->nameAndSignature;
```

Comment:
Assign this in the declaration, please. In fact all of these can be declared and assigned in order.